_ingest-pipelines/processors/dot-expander.md

```diff
@@ -0,0 +1,303 @@
+  "_id": "1",
+  "_version": 58,
+  "_seq_no": 57,
+  "_primary_term": 30,
+  "found": true,
+  "_source": {
+    "field": "Denver, CO"
+  }
+}
+```
+
+## Nested fields
+
+The `dot_expander` processor consolidates the `user.address.city` and `user.address.state` fields by merging with an existing `address`, `city`, and `state` field nested under `user`. If the field is a scalar value, then it will turn that field into an array. Take for example the following document:
+
+```json
+{
+    "dot_expander": {
+        "field": "user.address.city",
+        "override": true
+    }
+}
+```
+
+With `override:false` (or unset), the dot-pathed top-level field gets appended to the target object node, so it transforms into:
+
+```json
+{
+   "user": {
+        "address": {
+            "city": ["Denver", "Boulder"],
+            "state": "CO"
+        }
+    }
+}
+```
+
+With `override:true`, the dot-pathed top-level field value overwrites the target object node, so it transforms into:
+
+```json
+{
+   "user": {
+        "address": {
+            "city": "Boulder",
+            "state": "CO"
+        }
+    }
+}
+```
+
+If the field value is set to a wildcard `*`, the processor expands all top-level dotted field names, for example:  
+
+```json
+{
+    "dot_expander": {
+        "field": "*"
+    }
+}
+```
+
+Take for example the following document:
+
+
+```json
+{
+    "user.address.city": "Denver",
+    "user.address.state": "CO"
+}
+```
+
+The `dot_expander` processor transforms that document into the following fields, all nested under `user`:
+
+```json
+{
+    "user": {
+        "address": {
+            "city": "Denver"
+        }
+    },
+    "user": {
+        "address": {
+            "state": "CO"
+        }
+    }
+}
+```
+
+If a field is nested within a structure without dots, you can use the `path` parameter to traverse the non-dotted structure. For example, if you have the field `user.address`, you can use the `dot_expander processor` to expand it into an object field named `user` with a nested field named `address`, as shown in the following example: 
+
+```json
+{
+    "dot_expander": {
+        "path": "user.address",
+        "field": "*"
+    }
+}
+```
+
+Then take for example the following document: 
+
+```json
+{
+    "user": {
+        "address": {
+            "city.one": "Denver",
+            "city.two": "Houston",
+            "state.one": "CO",
+            "state.two": "TX"
+        }
+    }
+}
+```
+
+The `dot_expander` processor transforms the document into:
+
+```json
+{
+    "user": {
+        "address": {
+            "city": {
+                "one": "Denver",
+                "two": "Houston"
+            },
+            "state": {
+                "one": "CO",
+                "two": "TX"
+            }
+        } 
+    }
+}
+```
+
+To ensure proper expansion of the `user.address.city` and `user.address.state` fields and handle conflicts with pre-existing fields, use a similar configuration as the following document: 
```

Comment:
We need an example with an actual conflict to illustrate how conflicts are handled. First, an example of a simple document with a conflict, and next, an example with rename. Let me know if I can help with examples.